PowerFGT/Public/cmdb/vpn/ipsec/phase1-interface.ps1

```diff
@@ -3,6 +3,226 @@
+        [switch]$netdevice,
+        [Parameter (Mandatory = $false)]
+        [switch]$addroute,
+        [Parameter (Mandatory = $false)]
+        [switch]$autodiscoverysender,
+        [Parameter (Mandatory = $false)]
+        [switch]$autodiscoveryreceiver,
+        [Parameter (Mandatory = $false)]
+        [switch]$exchangeinterfaceip,
+        [Parameter (Mandatory = $false)]
+        [int]$networkid,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet('disable', 'on-idle', 'on-demand', IgnoreCase = $false)]
+        [string]$dpd,
+        [Parameter(Mandatory = $false)]
+        [ValidateRange(0, 10)]
+        [int]$dpdretrycount,
+        [Parameter(Mandatory = $false)]
+        [int]$dpdretryinterval,
+        [Parameter (Mandatory = $false)]
+        [switch]$idletimeout,
+        [Parameter (Mandatory = $false)]
+        [hashtable]$data,
+        [Parameter(Mandatory = $false)]
+        [String[]]$vdom,
+        [Parameter(Mandatory = $false)]
+        [psobject]$connection = $DefaultFGTConnection
+    )
+
+    Begin {
+    }
+
+    Process {
+
+        $invokeParams = @{ }
+        if ( $PsBoundParameters.ContainsKey('vdom') ) {
+            $invokeParams.add( 'vdom', $vdom )
+        }
+
+        $uri = "api/v2/cmdb/vpn.ipsec/phase1-interface"
+        $_interface = new-Object -TypeName PSObject
+
+        $_interface | add-member -name "name" -membertype NoteProperty -Value $name
+
+        $_interface | add-member -name "type" -membertype NoteProperty -Value $type
+        $_interface | add-member -name "interface" -membertype NoteProperty -Value $interface
+        $_interface | add-member -name "psksecret" -membertype NoteProperty -Value $psksecret
+
+        if ( $PsBoundParameters.ContainsKey('ikeversion') ) {
+            $_interface | add-member -name "ike-version" -membertype NoteProperty -Value $ikeversion
+        }
+
+        if ( $PsBoundParameters.ContainsKey('proposal') ) {
+            $_interface | add-member -name "proposal" -membertype NoteProperty -Value ($proposal -join " ")
+        }
+
+        if ( $type -eq "static" ) {
+            if ($PsBoundParameters.ContainsKey('remotegw')) {
+                $_interface | add-member -name "remote-gw" -membertype NoteProperty -Value $remotegw
+            }
+            else {
+                throw "You need to specify the remote-gw when use type static"
+            }
+        }
+        if ( $PsBoundParameters.ContainsKey('peertype') ) {
+            $_interface | add-member -name "peertype" -membertype NoteProperty -Value $peertype
+        }
+
+        if ( $PsBoundParameters.ContainsKey('dhgrp') ) {
+            $_interface | add-member -name "dhgrp" -membertype NoteProperty -Value ($dhgrp -join " ")
+        }
+
+        if ( $PsBoundParameters.ContainsKey('netdevice') ) {
+            if ($netdevice) {
+                $_interface | Add-member -name "net-device" -membertype NoteProperty -Value "enable"
+            }
+            else {
```

Comment:
The else should be on the primary If ? If netdevice is a parameter, you set its value to enable, else it's not a parameter so you set it to disable ?